inference/core/workflows/core_steps/common/utils.py

```diff
@@ -48,38 +50,72 @@ def attach_prediction_type_info(
     return predictions
 
 
-def attach_parent_info(
-    images: List[Dict[str, Any]],
+def convert_to_sv_detections(
     predictions: List[Dict[str, Any]],
-    nested_key: Optional[str] = "predictions",
-) -> List[Dict[str, Any]]:
-    return [
-        attach_parent_info_to_prediction(
-            image=image, prediction=prediction, nested_key=nested_key
-        )
-        for image, prediction in zip(images, predictions)
-    ]
+    predictions_key: str = "predictions",
+    keypoints_key: str = "keypoints",
+    detection_id_key: str = "detection_id",
+    parent_id_key: str = PARENT_ID_KEY,
+) -> List[Dict[str, Union[sv.Detections, Any]]]:
+    converted_predictions: List[Dict[str, Union[sv.Detections, Any]]] = []
+    for p in predictions:
+        converted_prediction = deepcopy(p)
+        detections = sv.Detections.from_inference(p)
+        if any(keypoints_key in d for d in p[predictions_key]):
```

Comment:
I am generally in favour of having `sv.Detections` for all detections-based use cases, but `supervision` itself seems to diverge from that idea. In the latest release we got new abstraction [sv.KeyPoints](https://github.com/roboflow/supervision/blob/develop/supervision/keypoint/core.py) which do not hold bboxes, and is invariant on "different skeleton for different class" topic - which makes it not 100% suitable for our usage now. 
I propose to keep as is for now, but we need to have discussion what to do next - definitely we need support for multiple skeletons (this is in general good change for `supervision` as this makes the abstraction more robust), but I would say that's suboptimal to loose info about Bounding Boxes when model provides them - maybe we could have both data outputs later in the future - one of Object Detection kind and another of Keypoints kind